File: StudyAkka/src/main/java/akka/app/actors/MapActor.java
package akka.app.actors;

import akka.actor.ActorRef;
import akka.actor.UntypedActor;
import akka.app.messages.MapData;
import akka.app.messages.WordCount;
import scala.actors.threadpool.Arrays;

import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;


/**
 * 对传进来的数据进行单词的切分
 * Created by liguodong on 2015/12/17.
 */
public class MapActor extends UntypedActor {

    private ActorRef reduceActor = null;

    String[] STOP_WORDS = {"a","is"};//这里面的单词不在单词统计的范围内

    private List<String> STOP_WORDS_LIST = Arrays.asList(STOP_WORDS);

    public MapActor(ActorRef inReduceActor){ reduceActor = inReduceActor;}


    //message 接收消息
    @Override
    public void onReceive(Object message) throws Exception{

        if(message instanceof String){//对消息类型进行判断
            String work = (String)message;
            //map the words in the sentence
            MapData data = evaluateExpression(work);//对单词进行切分
            //send the result to ReducerActor
            reduceActor.tell(data);//对处理的结果交给ReduceActor
        }else
            unhandled(message);//不在处理的范围
    }

    //单词切分工作
    private MapData evaluateExpression(String line){
        List<WordCount> dataList = new ArrayList<WordCount>();

        StringTokenizer parser = new StringTokenizer(line);//切分

        while(parser.hasMoreTokens()){
            String word = parser.nextToken().toLowerCase();//获取切分的元素再转化成小写
            if(!STOP_WORDS_LIST.contains(word)){
                dataList.add(new WordCount(word,Integer.valueOf(1)));
            }
        }
        return new MapData(dataList);
    }
}

File: StudyAkkaWithSBT/src/main/scala/akka/app/actors/MapActor.scala
package akka.app.actors

import java.util.{StringTokenizer, ArrayList}

import akka.actor.{ActorRef, Actor}
import akka.app.message.{Word, MapDada}

/**
  * Created by liguodong on 2015/12/19.
  */
class MapActor(reduceActor:ActorRef) extends Actor{

  val STOP_WORDS_LIST = List("a","is")

  def receive: Receive = {
    case message:String =>
      reduceActor ! evaluateExpression(message) //发送给reduceActor
    case _ =>
  }

  //数据切分
  def evaluateExpression(line:String):MapDada = {

    var dataList = new ArrayList[String]


    var parser:StringTokenizer = new StringTokenizer(line)//单词切分

    var defaultCount:Integer = 1
    while(parser.hasMoreTokens){
      var word:String = parser.nextToken().toLowerCase()
      if(!STOP_WORDS_LIST.contains(word)){
        //dataList.add(new Word(word,defaultCount))
        dataList.add(word)
      }
    }
    return new MapDada(dataList)
  }
}

